### PR TITLE
fix(groups): group icon render issues

### PR DESCRIPTION
**Pull request by josephmcg**
**What this PR does** 📖
- fix group icon render
- improve readability & scalability for toolbar render logic
- create some conversation hooks, switch toolbar and mobileToolbar to script setup() syntax
- improve flexibility of existing call method, allowing params. this is helpful in cases of calling from the sidebar, since you don't always just want to scrape conversation id from the route. These hooks could probably use some additional refactoring in that regard, but that can be handled separately 

**Which issue(s) this PR fixes** 🔨

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Review by aewing — APPROVED** [review]

**Comment by josephmcg** [comment]
working on composition api approach to reduce reused logic, moving to draft for now

File: components/compositions/conversations.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { computed, ComputedRef } from 'vue'
+import { Conversation } from '~/libraries/Iridium/chat/types'
+import iridium from '~/libraries/Iridium/IridiumManager'
+
+export function conversationHooks() {
```
Comment (aewing) [inline components/compositions/conversations.ts:5]: Why

Comment (josephmcg) [reply components/compositions/conversations.ts:5]: so any updates only have to happen in one file, rather than two

Comment (aewing) [reply components/compositions/conversations.ts:6]: what do we gain vs putting the equivalent in chat & webrtc managers instead? (`setActiveConversation`, etc...)

Comment (josephmcg) [reply components/compositions/conversations.ts:7]: that's a valid approach, I suppose I enjoy the separation of concerns.

The managers seem to be primarily concerned with iridium interaction and data manipulation. While these hooks are more about organizing data in a way that components can easily use it. I believe Vue has some special caching that goes on with computed values to improve performance

Comment (aewing) [reply components/compositions/conversations.ts:7]: the managers are concerned with the data and methods of protocol functions. that includes participants, active conversation, whether it's a group, etc...

Comment (josephmcg) [reply components/compositions/conversations.ts:7]: I'd like to keep the managers encapsulated if possible and avoid calling each other inside manager code. These hooks are resolving an issue of repeated computed properties across multiple components.

**Pull request merged**